src/controllers/campaign/stepThree.ts

```diff
@@ -0,0 +1,27 @@
+import { AppResponse, uploadSingleFile } from '@/common/utils';
+import AppError from '@/common/utils/appError';
+import { catchAsync } from '@/middlewares';
+import { campaignModel } from '@/models';
+import { Request, Response } from 'express';
+import { DateTime } from 'luxon';
+
+export const thirdStep = catchAsync(async (req: Request, res: Response) => {
+	const userId = req.user?._id;
+	const { story, id } = req.body;
+	const image = req.file;
+
+	if (!story || !id || !image) {
+		throw new AppError('All fields are required!', 400);
+	}
+	const dateInMilliseconds = DateTime.now().toMillis();
+	const fileName = `${userId}/campaign-image/${userId}-${dateInMilliseconds}.${image.mimetype.split('/')[1]}`;
```

Comment:
scope the image folder in a more organized format 

userid
 campaigns 
    campaignID
       ... store images here...
       
       so 
       `${userId}/campaigns/${campaignId}/${dateInMilliseconds}.${image.mimetype.split('/')[1]}`;